addons/website_event/static/src/snippets/options.js

```diff
@@ -58,6 +61,12 @@ options.registry.WebsiteEvent = options.Class.extend({
         }
         return this._super(...arguments);
     },
+    /**
+     * @override
+     */
+    _computeWidgetVisibility() {
+        return this.modelName === 'event.event';
+    },
```

Comment:
No params :thinking: ?
Don't you need to condition it based on the widget name?

Reply:
Changed it to _computeVisibility as the entire logic of the option relies on the model being event.event, so if it's not an event.event, the option should be hidden entirely.

Reply:
Shouldn't we patch the `snippets.editor.js` to allow to add the correct selector here then?
https://github.com/odoo/odoo/blob/master/addons/website_event/views/snippets/snippets.xml#L73 -> selector that would include html[mo=event]
I guess the system doesn't allow that because it searches on html children, an addBack (or equivalent) would allow that, and would be cleaner, no?

@qsm-odoo You decide about that

Reply:
> I guess the system doesn't allow that because it searches on html children, an addBack (or equivalent) would allow that, and would be cleaner, no?

The system does allow that automatically :wink: So yes, it would be better to not instantiate the option at all if the visibility condition is not meant to change during edition if possible :+1:

Reply:
That's even better then :+1: 
Whatever it is working or not anyway, that's the way to go, it will be better to hook the JS to the correct part of the DOM, in this case depending of the main object attribute

@detrouxdev I'll let you have a look